File: src/Cell.cpp

#include "Cell.h"
#include <iostream>

using namespace std;


Cell::Cell(int x, int y, char v): value(v), xCord(x), yCord(y){}
Cell::Cell(): value('X'), xCord(0), yCord(0) {}


char Cell::getValue() const {
	return value;
}


void Cell::setValue(char value) {
	this->value = value;
}

bool Cell::isSameValue(char c) {
	return (c == this->value);
}


void Cell::printCell() {
	cout << "(" << this->xCord << "," << this->yCord << ")";
}


int Cell::getXCord() const {
	return this->xCord;
}


int Cell::getYCord() const {
	return this->yCord;
}


bool Cell::sameCoord(Cell c) {
	if (c.getXCord() == getXCord()) {
		return (c.getYCord() == getYCord());
	}
	return false;
}

bool Cell::operator <(const Cell &compareCell) const {
	if(this->getXCord() == compareCell.getXCord()) {
		return (this->getYCord() < compareCell.getYCord());
	} else {
		return (this->getXCord() < compareCell.getXCord());
	}
}

bool Cell::operator ==(const Cell& compareCell) const {
	if(this->getXCord() == compareCell.getXCord()) {
		return (this->getYCord() == compareCell.getYCord());
	} else {
		return false;
	}
	
	
}

File: src/Cell.java


public class Cell implements Comparable {
    int xCord, yCord;
    char value;

    public Cell(int x, int y, char v) {
        xCord = x;
        yCord = y;
        value = v;
    }

    public char getValue() {
        return this.value;
    }

    public void setValue(char v) {
        this.value = v;
    }

    public boolean isSameValue(char c) {
        return (c == this.value);
    }


    public void printCell() {
        System.out.print("(" + this.xCord + "," + this.yCord + ")");
    }


    public int getXCord() {
        return this.xCord;
    }


    public int getYCord() {
        return this.yCord;
    }

    public boolean equals(Object compareCell) {
        if(this.xCord == ((Cell)compareCell).getXCord() && this.yCord == ((Cell)compareCell).getYCord()) {
            return true;
        }
        return false;
    }

    public boolean lessThan(Cell compareCell) {
        //If this.x is less, or x is equal and this.y is less
        if(this.xCord < ((Cell)compareCell).getXCord() ||
                (this.xCord == ((Cell)compareCell).getXCord() && this.yCord < ((Cell)compareCell).getYCord())) {
            return true;
        }
        return false;
    }

    @Override
    public int compareTo(Object compareCell) {
        if(this.xCord == ((Cell)compareCell).getXCord() && this.yCord == ((Cell)compareCell).getYCord()) {
            return 0;
        }

        //If this.x is less, or x is equal and this.y is less
        if(this.xCord < ((Cell)compareCell).getXCord() ||
                (this.xCord == ((Cell)compareCell).getXCord() && this.yCord < ((Cell)compareCell).getYCord())) {
            return -1;
        }
        return 1;
    }
}
